iterm2-shell-integration.plugin.zsh

```diff
@@ -0,0 +1,8 @@
+# Automagically add the utilities directory to $PATH
+
+# Add utilities to our path
+PLUGIN_BIN="$(dirname $0)/utilities"
+export PATH=${PATH}:${PLUGIN_BIN}
+
+# And source the helper functions, too.
+source "${0:h}/shell_integration/zsh"
```

Comment:
Since you're using the base-dir expansion `${0:h}` for sourcing, it'd be more consistent  to use that as well when assigning `PATH`. Also could potentially be a tiny bit slower to call `dirname` as it's a sub-shell call. But I could be wrong here.

Here's my suggestion:

```zsh
path+=("${0:h}/utilities")
```
You don't need to export, and this way you append to the tied array version of `$PATH`

Reply:
Just because I've too much time at hand, I wanted to check how many millis one could scrape off with the parameter expansion over the `dirname` call. Turns out quite a few, though negligable in it's direct effect:

```
$ hyperfine --runs 500 'echo $(dirname $PWD)' 'echo "${PWD:h}"'                                                                          

Benchmark 1: echo $(dirname $PWD)
  Time (mean ± σ):       5.3 ms ±   2.0 ms    [User: 0.5 ms, System: 1.9 ms]
  Range (min … max):     1.2 ms …  26.0 ms    500 runs

  Warning: Command took less than 5 ms to complete. Note that the results might be inaccurate because hyperfine can not calibrate the shell startup time much more precise than this limit. You can try to use the `-N`/`--shell=none` option to disable the shell completely.
  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet system without any interferences from other programs. It might help to use the '--warmup' or '--prepare' options.

Benchmark 2: echo "${PWD:h}"
  Time (mean ± σ):       0.6 ms ±   0.9 ms    [User: 0.0 ms, System: 0.1 ms]
  Range (min … max):     0.0 ms …   6.5 ms    500 runs

  Warning: Command took less than 5 ms to complete. Note that the results might be inaccurate because hyperfine can not calibrate the shell startup time much more precise than this limit. You can try to use the `-N`/`--shell=none` option to disable the shell completely.
  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet system without any interferences from other programs. It might help to use the '--warmup' or '--prepare' options.

Summary
  echo "${PWD:h}" ran
    8.98 ± 14.07 times faster than echo $(dirname $PWD)
```

Reply:
Your way is better, @nunico, I've switched to using it.

Reply:
and renamed the plugin file